src/org/linphone/setup/CDNProviders.java

```diff
@@ -140,62 +141,72 @@ public boolean updateProviders(String jsonString, boolean save) {
 
 
 	}
-
+	private static final Object locker = new Object();
 	private void updateProviders(JSONArray jsonArray) {
-		providers.clear();
-		g.domain_image_hash=new HashMap<String, Uri>();
-		Provider tmp;
-		for (int i = 0; i < jsonArray.length(); i++) {
-			tmp = new Provider();
-			try {
-
-				//change domains to valid domains, if not listed on the server that way
-				if(((JSONObject) jsonArray.get(i)).getString("domain").equals("zvrs.vatrp.net")){
-					tmp.domain="208.94.16.87";
-				}else if(((JSONObject) jsonArray.get(i)).getString("domain").equals("psip-lb.staging.purple.us")){
-					tmp.domain="psip-lb.test.purple.us";
-				}else {
-					tmp.domain = ((JSONObject) jsonArray.get(i)).getString("domain");
-				}
+		synchronized (locker) {
+			providers.clear();
+			g.domain_image_hash = new HashMap<String, Uri>();
+			Provider tmp;
+			for (int i = 0; i < jsonArray.length(); i++) {
+				tmp = new Provider();
+				try {
 
-				tmp.name = ((JSONObject) jsonArray.get(i)).getString("name");
-				tmp.icon = ((JSONObject) jsonArray.get(i)).getString("icon");
-				tmp.icon2x = ((JSONObject) jsonArray.get(i)).getString("icon2x");
-				//tmp.port=srvLookup
+					//change domains to valid domains, if not listed on the server that way
+					if (((JSONObject) jsonArray.get(i)).getString("domain").equals("zvrs.vatrp.net")) {
+						tmp.domain = "208.94.16.87";
```

Comment:
Get rid of this. This is hardcoded and pulled from the CDN. Should not be modified in app.